lib/cpp/src/thrift/concurrency/Mutex.h

```diff
@@ -101,25 +101,6 @@ class ReadWriteMutex {
   boost::shared_ptr<impl> impl_;
 };
 
-/**
- * A ReadWriteMutex that guarantees writers will not be starved by readers:
- * When a writer attempts to acquire the mutex, all new readers will be
- * blocked from acquiring the mutex until the writer has acquired and
- * released it. In some operating systems, this may already be guaranteed
- * by a regular ReadWriteMutex.
- */
-class NoStarveReadWriteMutex : public ReadWriteMutex {
-public:
-  NoStarveReadWriteMutex();
-
-  virtual void acquireRead() const;
-  virtual void acquireWrite() const;
-
-private:
-  Mutex mutex_;
-  mutable volatile bool writerWaiting_;
-};
```

Comment:
It's in a "public" header (kind of).
Are you sure you delete this without deprecation warning etc ?

In this file, ReadWriteMutex and RWGuard are not used inside Thrift either.
How about removing or deprecating all the three classes in a separate issue,
since these are somewhat more involved compared to private volatile removal ?